tools/topology/topology2/include/components/aec.conf

```diff
@@ -0,0 +1,160 @@
+#
+# Common widget aec
+#
+# A generic aec widget. All attributes defined herein are namespaced
+# by alsatplg to "Object.Widget.aec.N.attribute_name"
+#
+# Usage: this component can be used by instantiating it in the parent object. i.e.
+#
+# 	Object.Widget.aec."N" {
+# 		index		1
+# 		type		"effect"
+#	}
+#
+# Where N is the unique instance number for the aec object within the same alsaconf node.
+Define {
+	AEC_UUID	"a6:a0:80:b7:9f:26:6f:46:b4:77:23:df:a0:5a:f7:58"
+}
+
+Class.Widget."aec" {
+	#
+	# Pipeline ID for the aec widget object
+	#
+	DefineAttribute."index" {}
+
+	#
+	# aec object instance
+	#
+	DefineAttribute."instance" {}
+
+	#include common component definition
+	<include/components/widget-common.conf>
+
+	DefineAttribute."num_audio_formats" {
+		# Token set reference name and type
+		token_ref	"sof_tkn_comp.word"
+	}
+
+	#
+	# aec widget UUID
+	#
+	DefineAttribute."uuid" {
+		type	"string"
+		# Token set reference name and type
+		token_ref	"sof_tkn_comp.uuid"
+	}
+
+	DefineAttribute."cpc" {
+		# Token set reference name and type
+		token_ref	"sof_tkn_comp.word"
+	}
+
+	DefineAttribute."is_pages" {
+		# Token set reference name and type
+		token_ref	"sof_tkn_comp.word"
+	}
+
+	# Attribute categories
+	attributes {
+		#
+		# The aec widget name would be constructed using the index and instance attributes.
+		# For ex: "aec.1.1" or "aec.10.2" etc.
```

Comment:
no. we absolutely want to avoid generic terms such as 'aec'. It has to be clearly linked with the 3rd party or library used.

Reply:
sure,thanks